presto-main/src/main/java/io/prestosql/server/ui/FormWebUiAuthenticationManager.java

```diff
@@ -371,15 +357,30 @@ private static String parseJwt(byte[] hmac, String jwt)
                 .getSubject();
     }
 
-    private static Optional<Integer> getForwarderPort(HttpServletRequest request)
+    private static HttpUriBuilder toUriBuilderWithForwarding(HttpServletRequest request)
     {
-        if (!isNullOrEmpty(request.getHeader(X_FORWARDED_PORT))) {
-            try {
-                return Optional.of(parseInt(request.getHeader(X_FORWARDED_PORT)));
-            }
-            catch (ArithmeticException ignore) {
+        HttpUriBuilder builder;
+        if (isNullOrEmpty(request.getHeader(X_FORWARDED_PROTO)) && isNullOrEmpty(request.getHeader(X_FORWARDED_HOST))) {
```

Comment:
When did we decide that the `X_FORWARDED_HOST` is not an "activator".  We decided that `X_FORWARDED_PROTO` alone without the other two doesn't make much sense.  I am fine making it an activator also, since none of the LBs seem to do that, so it shouldn't matter.  BTW the port header isn't part of the spec.  The `X_FORWARDED_HOST` header is documented here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host

Reply:
It's probably best to make any of these an activator. I don't see a downside in doing so.

Reply:
I think this should be
```java
.hostAndPort(HostAndPort.fromString(request.getHeader(HOST)))
```
We don't want to add the port number unless the original request had one. Though, what happens if there's no `Host` header? Is that legal?